# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/A7mad7-7/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

In [2]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


Lane: Refresh / Content Opportunity Scoring

Why this lane:
Our team faces a critical capacity constraint: human editors can only manually review and update a small batch of pages per week. Creating brand-new content is resource-intensive, whereas optimizing existing high-value assets yields faster and higher return on investment (ROI). Instead of randomly selecting pages or relying solely on static rules, this project focuses on building an intelligent decision-support system. It ranks content by refresh priority, surfaces decaying pages that still retain active search demand, and attaches clear reason codes so editors can intervene at the right time—maximizing impact without wasting team capacity.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


The Decision: Which pages should be placed at the top of our weekly content review queue for immediate optimization versus monitoring or leaving alone?The Action: Content editors will review the top-ranked candidate pages, update outdated information, expand thin content, or fix metadata based on attached reason codes.Unit of Analysis: A single content item/page per client (content_id).Cost of a Wrong Call:False Positive (Ranking a fine page too high): Wastes scarce editing capacity on a page that didn't need intervention. More critically, it displaces a truly critical page from the review queue.False Negative (Missing a critically decaying page): Causes high-demand content to lose ranking authority, traffic, and client trust due to delayed action.Why ML helps: Simple static rules fail to balance multiple factors (e.g., search volume vs. freshness decay vs. engagement rates) simultaneously. ML optimizes the top-$K$ decision boundary, ensuring limited editing capacity yields maximum traffic protection.  

## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [3]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load the starter anonymized dataset
df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

# Filter dataset based on standard criteria (matured pages with search presence)
df_filtered = df[(df["impressions_90d"] > 0) & (df["content_age_days"] >= 90)].copy()

total_pages = len(df_filtered)

# Number 1: Total scale of declining pages
declining_pages = len(df_filtered[df_filtered["trend_direction"] == "down"])
declining_pct = (declining_pages / total_pages) * 100

# Number 2: High-demand pages in decline (high impact opportunity)
high_demand_declining = len(
    df_filtered[(df_filtered["trend_direction"] == "down") & (df_filtered["impressions_90d"] >= 500)]
)

# Number 3: Freshness risk (pages not updated in > 180 days with solid impressions)
stale_visible_pages = len(
    df_filtered[(df_filtered["days_since_last_update"] >= 180) & (df_filtered["impressions_90d"] >= 500)]
)
stale_visible_pct = (stale_visible_pages / total_pages) * 100

# Print the verifiable real numbers
print(f"Total analyzed pages: {total_pages:,}")
print(f"1. Declining pages: {declining_pages:,} ({declining_pct:.1f}%)")
print(f"2. High-demand declining pages (>=500 imp): {high_demand_declining:,}")
print(f"3. Stale high-visibility pages (>=180d untouched, >=500 imp): {stale_visible_pages:,} ({stale_visible_pct:.1f}%)")

Total analyzed pages: 30,000
1. Declining pages: 16,262 (54.2%)
2. High-demand declining pages (>=500 imp): 9,961
3. Stale high-visibility pages (>=180d untouched, >=500 imp): 17 (0.1%)


Key Data Evidence Supporting This Lane:

1. Problem Scale: Out of the analyzed portfolio of 30,000 pages, 16,262 pages (54.2%) are currently in a downward performance trend.
2. High-Impact Opportunity: There are 9,961 declining pages that still retain strong demand (>= 500 impressions in 90 days), representing immediate high-value targets for content recovery.
3. Actionable Driver (Freshness Decay): We identified 17 high-visibility pages (>= 500 impressions) that have gone over 180 days without an update (0.1%), representing an initial subset of neglected, high-performing content.

## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


What this work CAN claim:

Decision-Support & Prioritization: It provides an objective, data-driven framework to rank content review queues, reducing manual inspection overhead for editors.

Directional Signals: It identifies observable patterns, correlated risks (e.g., freshness decay, engagement drops), and directional performance trends within historical data.

Resource Efficiency: It optimizes capacity allocation by focusing limited human effort on high-impact, declining pages.

What this work CANNOT claim:

No Causal Proof: It cannot prove that lack of updates directly caused a traffic drop (correlation does not equal causation).

No 'Predicting Google': It does not reverse-engineer or predict Google's proprietary ranking algorithm; it only measures past observable metrics.

No Guaranteed Recovery: Updating a high-priority page does not guarantee a 100% traffic recovery, as external market shifts and competitor actions also influence rankings.

## Self-check

Before you submit, confirm each line honestly:

- [x] Every section above is filled — markdown thinking AND the code that backs it
- [x] The notebook runs top to bottom with no errors (Runtime → Run all)
- [x] No client names, URLs, or private queries anywhere
- [x] My claims use careful words: observed, measured, directional, decision-support
- [x] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.